### ISSUE TITLE
Fix Mapper 342 

### ISSUE THREAD

**Issue by Oskar2024** [TASEmulators/fceux#638]
![231420106-2dccc92b-cde2-4cd3-8d06-c6729e2a9225](https://user-images.githubusercontent.com/131452546/233832161-a386380f-ae66-4f6e-9ddd-2e24025a63d7.jpg)
![231420100-06c07e9b-5016-4dce-8a41-1c60b1320c5f](https://user-images.githubusercontent.com/131452546/233832163-6f159e45-4857-4cf8-9539-067983ffd8ad.jpg)
![231420089-a58c87d8-085d-44b6-821f-61cab9cb288b](https://user-images.githubusercontent.com/131452546/233832164-06c86b52-da1e-408a-9800-c401f2876b7d.jpg)
 - OS and Version: [e.g. Windows 10]
 - Version of FCEUX used (2.6.5)
 
when running mapper 342 from coolhirl mapper games do not work properly launching Crisis Force from mapper 342/coolgirl build glitch image in game

**Comment by negativeExponent** [comment]
in this same rom, the game Castlevania:Special also has CHR issues

**Comment by ClusterM** [comment]
> 1. in this same rom, the game Castlevania:Special also has CHR issues. most probably translation for CHR address on mapper 24 mode is incorrect.
> 
> Arkanoid II from 320-in-1 rom variant also broken - flags check on mapper 18 probably incorrect

Arkanoid II works fine on my multicart ROMs. It uses mapper [152](https://nescartdb.com/search/advanced?ines=152), not 18. Can you contact me on Discord and share the ROM? My username: cluster#1750

> possible typo in https://github.com/TASEmulators/fceux/blame/master/src/boards/coolgirl.cpp#L886-L887 line indicates it should apply on prg_bank_a, but it was applied to chr_bank_a instead. must be a typo

You are right, thank you. Fixed.

> 4. the message "Cycle IRQ mode is not supported, please report to Cluster" can spam over and over causing emulator to be unresponsive (check Crisis Force). Cycle IRQ mode 

Original Crisis Force doesn't use cycle IRQ and works fine for me. There are no official games that use this mode. Even FCEUX started supporting it only since a certain commit. Seems like it's some hack. I tried to run this game from "COOLGIRL 7-IN-1(MMK-02E-00) .unf" on a **real** hardware, and it doesn't work properly:

![image](https://user-images.githubusercontent.com/4236181/236790467-9cafeab0-bd89-47df-ada3-d810e8d91ea8.png)
(i really need to clean the dust)

> 5. possible missing "break" statement
>    https://github.com/TASEmulators/fceux/blob/master/src/boards/coolgirl.cpp#L1462-L1465C1

Fixed, thank you!

**Comment by ClusterM** [comment]
`Super Konami 8-in-1(Yhc-4006-00) (Unl).unf` uses wrong flags for VRC2/VRC4 mappers when starts Castlevania Special, so it's a bad ROM.
Same problem with a Crisis Force. Seems like those flags are messed up in the other emulators (because of my bad documentation) and those multigame ROMs were built for such emulators, not a real hardware.

I fixed mapper documentation: https://www.nesdev.org/wiki/NES_2.0_Mapper_342

In ROM `Super Konami 8-in-1(Yhc-4006-00) (Unl) [f].unf`:
* Change byte at offset `0x1E5B4` from `0x18` to `0x98` to run Castlevania Special correctly.
* Change byte at offset `0x1E5B5` from `0x18` to `0x98` to run Crisis Force correctly.

**Comment by negativeExponent** [comment]
>Arkanoid II works fine on my multicart ROMs. It uses mapper [152](https://nescartdb.com/search/advanced?ines=152), not 18. Can you contact me on Discord and share the ROM? My username: cluster#1750

true. Arkanoid does use NES mapper 152 (i was referring to the mapper code provided with coolgirl's mapper registers, aka **coolgirl_mapper_reg=18 = NES mapper #152**)

>Super Konami 8-in-1(Yhc-4006-00) (Unl).unf uses wrong flags for VRC2/VRC4 mappers when starts Castlevania Special, so it's a bad ROM.

does this mean that all, if not most multicart dumps which uses coolgirl_mapper_24(iNES Mapper #23), flag 0 are bad dumps?

**Comment by ClusterM** [comment]
And documentation was not correct too:(

**Comment by ClusterM** [comment]
And documentation was not correct too:(

**Comment by YingHuoChong4913** [comment]
> Collaborator

This is the ROM I made three years ago, and it was also run with the emulator three years ago, there was no problem at that time, the later version did have problems, I also tested it myself, it is true. I will fix it later, and thank ClusterM for the repair.
